### PR TITLE
Save time for selected date

### LINKED ISSUE #606: Reset time in selected date after changes
Now, time in selected date resets after any changes. It's very uncomfortable if you are using datepicker with timepicker. I'm resolved this bug and wrote tests to check it [here](https://github.com/zouxuoz/react-datepicker). Can I send pull request?

### PR DESCRIPTION
**Pull request by zouxuoz**
resolve #606

### CODE REVIEW

File: src/datepicker.jsx [hunk 1]
```diff
@@ -143,6 +143,14 @@ var DatePicker = React.createClass({
 
   setSelected (date, event) {
     if (!isSameDay(this.props.selected, date)) {
+      if (this.props.selected) {
+        date.set({
```
Comment (rafeememon) [inline src/datepicker.jsx]: Let's avoid mutating the moment object by creating a new object based on `date`

Comment (zouxuoz) [inline src/datepicker.jsx]: @rafeememon fixed

**Comment by zouxuoz** [comment]
@rafeememon what should I change for merging this pull request?

**Comment by rafeememon** [comment]
lgtm, thanks!

**Pull request merged**